aws/resource_aws_autoscaling_group_test.go

```diff
@@ -1007,10 +1003,11 @@ func TestAccAWSAutoScalingGroup_initialLifecycleHook(t *testing.T) {
 					testAccCheckAWSAutoScalingGroupHealthyCapacity(&group, 2),
 					resource.TestCheckResourceAttr(
 						"aws_autoscaling_group.bar", "initial_lifecycle_hook.#", "1"),
-					resource.TestCheckResourceAttr(
-						"aws_autoscaling_group.bar", "initial_lifecycle_hook.391359060.default_result", "CONTINUE"),
-					resource.TestCheckResourceAttr(
-						"aws_autoscaling_group.bar", "initial_lifecycle_hook.391359060.name", "launching"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs("aws_autoscaling_group.bar", "initial_lifecycle_hook.*", map[string]string{
+						"default_result": "CONTINUE",
+						"name":           "launching",
+					}),
+					// TODO: TypeSet hash
 					testAccCheckAWSAutoScalingGroupInitialLifecycleHookExists(
 						"aws_autoscaling_group.bar", "initial_lifecycle_hook.391359060.name"),
```

Comment:
should we still keep `DisableBinaryDriver: true,` to be able to support this check for the time being?

Reply:
Yes we could, I will switch it back

Reply:
oh ok! wasn't sure if it would cause problems while running the tests in the binary driver testing environment in TC

Reply:
the test would just error if we switched to V2 of the SDK today is all. `DisableBinaryDriver` is not an option for V2 of the SDK, but keeping it set to true is a good breadcrumb for me (even with the TODO)